repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/filters/tenantculling/TenantCullingFilterStandaloneFunctionalTest.groovy

```diff
@@ -0,0 +1,118 @@
+
+        where:
+        testName                           | headers
+        'Unencoded tenant-to-roles map'    | [(TENANT_ROLES_MAP): '{"defaultTenant": [], "tenant1": ["role1", "role2"], "tenant3": ["role3"]}']
+        'Invalid JSON tenant-to-roles map' | [(TENANT_ROLES_MAP): Base64.encoder.encodeToString('"defaultTenant": [], "tenant1": ["role1", "role2"], "tenant3": ["role3"]}'.bytes)]
+        'Garbage tenant-to-roles map'      | [(TENANT_ROLES_MAP): 'u832hrh349tb31>{}_+{}']
+        'No tenant-to-roles map'           | null
+    }
+
+    def "culls all tenants when no relevant roles exist"() {
+        given: "a tenant-to-roles map"
+        Map tenantToRolesMap = ["defaultTenant": [], "tenant1": ["role1", "role2"], "tenant3": ["role3"]]
+
+        when: "the request is made"
+        MessageChain mc = deproxy.makeRequest(
+            url: reposeEndpoint,
+            method: 'GET',
+            headers: [(TENANT_ROLES_MAP): encodeTenantToRolesMap(tenantToRolesMap)]
+        )
+
+        then: "the origin service should not receive any tenants"
+        mc.handlings.size() == 1
+        mc.receivedResponse.code as Integer == SC_OK
+        mc.handlings[0].request.headers.findAll(TENANT_ID).isEmpty()
+        decodeTenantToRolesMap(mc.handlings[0].request.headers.findAll(TENANT_ROLES_MAP).first()).isEmpty()
+    }
+
+    def "sends multiple tenants matching multiple relevant roles"() {
+        given: "a tenant-to-roles map"
+        Map tenantToRolesMap = ["defaultTenant": [], "tenant1": ["role1", "role2"], "tenant3": ["role3"]]
+
+        when: "the request is made"
+        MessageChain mc = deproxy.makeRequest(
+            url: reposeEndpoint,
+            method: 'GET',
+            headers: [
+                (RELEVANT_ROLES)  : 'role1,role3',
+                (TENANT_ROLES_MAP): encodeTenantToRolesMap(tenantToRolesMap)
+            ]
+        )
+
+        then: "the origin service should not receive any tenants"
+        mc.handlings.size() == 1
+        mc.receivedResponse.code as Integer == SC_OK
+
+        when: "the origin service request headers are parsed"
+        HeaderCollection requestHeaders = mc.handlings[0].request.headers
+        List originTenantIds = requestHeaders.findAll(TENANT_ID).collect { it.split(",") }.flatten()
+        Map originTenantToRolesMap = decodeTenantToRolesMap(requestHeaders.findAll(TENANT_ROLES_MAP).first())
+
+        then: "the origin service should have received relevant tenants and roles"
+        originTenantIds == ['tenant1', 'tenant3']
+        originTenantToRolesMap == (tenantToRolesMap - ['defaultTenant': []])
+    }
+
+    Map decodeTenantToRolesMap(String encodedTenantToRolesMap) {
+        jsonSlurper.parse(Base64.decoder.decode(encodedTenantToRolesMap)) as Map<String, Set<String>>
+    }
+
+    String encodeTenantToRolesMap(Map tenantToRolesMap) {
+        Base64.encoder.encodeToString(JsonOutput.toJson(tenantToRolesMap).bytes)
+    }
```

Comment:
These Base64 En/Decoders can be replaced by using the methods in:
> `org.openrepose.commons.utils.string.Base64Helper`

Reply:
Looking at the helper, it doesn't go as far as i was suggesting we take it. We have multiple filters that deal with the map at this point i was suggesting that we make helper that encodes and decodes the whole map. Probably go ahead and move the type alias to the same object.